### ISSUE TITLE
Unable to use "browser", "chrome", "firefox", or "adobe reader" as help viewers.

### ISSUE THREAD

**Issue by Jacksbabypig** [gap-system/gap#1579]
When I use ?SetHelpViewer to look for the recommended choices for help viewers, Chrome, Firefox and Adobe Reader are listed as options, yet when I try to use them I get the following:

```
gap> SetHelpViewer("chrome");
#I  GAP's help system has NO support for help viewer chrome!
```

and identical messages for Firefox and Adobe Reader.

Similarly, if I instead try SetHelpViewer("browser") then the command goes through just fine, but as soon as I attempt to open any help articles I get the following:

```
gap> SetHelpViewer("browser");
gap> ?SetHelpViewer
Help: Showing `Reference: SetHelpViewer'
Error, the name of the help file /proc/cygdrive/C/gap4r8/doc/ref/chap2.html#X87C1BFB2826488B0 must start with /cygdrive called from
HELP_VIEWER_INFO.(viewer).show( data ); at /proc/cygdrive/C/gap4r8/lib/helpbase.gi:785 called from
HELP_PRINT_MATCH( i ); at /proc/cygdrive/C/gap4r8/pkg/Browse/lib/helpoverwrite.g:68 called from
HELP_SHOW_MATCHES( books, str, true ) at /proc/cygdrive/C/gap4r8/lib/helpbase.gi:1242 called from
<function "HELP">( <arguments> )
 called from read-eval loop at line 11 of *stdin*
```

Using 32-bit GAP installed via the .exe file on the main website on Windows 10. Same happens with the 64-bit version.
```

 ┌───────┐   GAP 4.8.7, 24-Mar-2017, build of 2017-03-24 22:43:07 (GMTST)
 │  GAP  │   https://www.gap-system.org
 └───────┘   Architecture: i686-pc-cygwin-gcc-default32
 Libs used:  gmp, readline
 Loading the library and packages ...
 Components: trans 1.0, prim 2.1, small* 1.0, id* 1.0
 Packages:   AClib 1.2, Alnuth 3.0.0, AtlasRep 1.5.1, AutPGrp 1.8, Browse 1.8.6, CRISP 1.4.4, Cryst 4.1.12,
             CrystCat 1.1.6, CTblLib 1.2.2, FactInt 1.5.4, FGA 1.3.1, GAPDoc 1.5.1, IO 4.4.6, IRREDSOL 1.3.1,
             LAGUNA 3.7.0, Polenta 1.3.7, Polycyclic 2.11, RadiRoot 2.7, ResClasses 4.6.0, Sophus 1.23, SpinSym 1.5,
             TomLib 1.2.6, Utils 0.46

```

**Comment by fingolfin** [comment]
Looking at the code, it turns out that help viewers "chrome" etc. you list are for Unix operating systems (except OS X). 

On Windows, the only available method seems to be (as you discovered) "browser", which is supposed to open the help in the default browser.

Could you please try to copy&paste the following into a GAP session, then try again with the help viewer set to "browser"?
```gap
  HELP_VIEWER_INFO.browser := rec(
  type := "url",
  show := function( filename )
    local pos, winfilename;
    if not StartsWith(filename, "/cygdrive") and
       not StartsWith(filename, "/proc/cygdrive") then
      Error( "the name of the help file ", filename , " must start with /cygdrive or /proc/cygdrive" );
    else
      winfilename:=MakeExternalFilename( SplitString( filename, "#" )[1] );
    fi;
    Print( "Opening help page ", winfilename, " in default windows browser ... \c" );
    Exec( Concatenation("start ", winfilename ) );
    Print( "done! \n" );         
  end
  );
```

**Comment by alex-konovalov** [comment]
@Jacksbabypig thank you for reporting this. 

@fingolfin, I can reproduce the problem on Windows 7, and confirm that your fix works. 

I think on the Windows 8 machine, which I've used to check manually that the help system works, the path is reported as starting with `/cygdrive`, so I was not able to observe this.

We should fix this in GAP 4.8.8. @fingolfin, will you submit a PR to the stable-4.8, or shall I submit it?

**Comment by fingolfin** [comment]
@alex-konovalov feel free to take my proposed patch and turn it into a PR

**Comment by alex-konovalov** [comment]
Fixed in #1592
